partner_address_two_lines/models/res_partner.py

```diff
@@ -9,8 +9,10 @@ class ResPartner(models.Model):
 
     def _get_contact_name(self, partner, name):
         if self.env.context.get("_two_lines_partner_address"):
-            return "{}\n {}".format(
-                partner.commercial_company_name or partner.sudo().parent_id.name, name
-            )
-        else:
-            return super()._get_contact_name(partner, name)
+            company_name = partner.commercial_company_name
+            partner_name = partner.name
+            if company_name and partner_name:
+                # Only display two lines if both values are found,
+                # otherwise revert to the standard behavior.
```

Comment:
AFAIU if there's no name on the contact the "standard behaviour" will still display it as:
> Company name, address name
instead of 
> Company name
> address name

Don't we want do display only "Company name" on a single line here?

Reply:
If partner.name is not found, only the company name / name of the parent of the partner id will be displayed:
```    
def _get_contact_name(self, partner, name):
        return "%s, %s" % (partner.commercial_company_name or partner.sudo().parent_id.name, name)
```

Where parameter `name` corresponds to partner.name, which will be empty (the comma is stripped afterwards).

So basically, if we find both a company name and a partner name, we display it in two rows. If there's no partner name, then we keep the one-liner as it is now.

If you can come up with a more elegant solution, I'll be glad to test it!

Reply:
It might be ok, but please add a test to ensure it is the correct behaviour. :pray:

Reply:
Of course, tests added.

Reply:
I'm wondering, can't we strip the `\n` afterwards as it's done with the `,`?